src/server/node_manager.c

```diff
@@ -7111,6 +7111,25 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 				return (PBSE_UNKNODE);
 			}
 
+                        if ((pnode->nd_state & (INUSE_DOWN | INUSE_STALE)) && (svr_init == FALSE)) {
+
+				if(objtype == JOB_OBJECT) {
+                                   free(phowl);
+                                   free(execvncopy);
+                                   return (PBSE_BAD_NODE_STATE);
+				} 
+				else {
+					if(presv->ri_qs.ri_state == RESV_UNCONFIRMED) {
```

Comment:
Thanks, @prakashcv13 for your reviews.

We are handling advance & standing reservation differently since in general advance reservations gets rejected if enough nodes are not available to allocate. If it is next occurrence of standing reservations (already confirmed once, but degraded due to node down), the server will maintain in DEGRADE state to honor in next scheduling cycle. 

Please let me know if you think otherwise.

Reply:
The more I think on it, I feel degrading is the right way to go forward for advance as well as standing reservations - below is my rationale behind suggesting that -
All occurrences of a standing reservation do not get allocated the same set of nodes (there are plans to make them so, but as of now it is not necessary that all occurrences get the same set of nodes).
When set_nodes() is called from req_confirmresv() for a standing reservation, then you will be rejecting all the occurrences of the standing reservation even though the upcoming occurrences might be on the nodes that are UP at this time.
For advance reservation we do reject them if we do not have enough nodes, but at this point of time, we only know that the one of the nodes that scheduler allocated is down, there is a possibility that there are other nodes that can replace the down node and the reservation can still get confirmed. If we simply reject the scheduler's request we are not allowing the scheduler to replace the node that just went down.

Reply:
ok, Thanks @prakashcv13 
I will consider your thoughts on the next common approach for both cases.